### PR TITLE
slicer endpoint should not hang on new slicer creation resolves #659 …

### LINKED ISSUE #659: /txt/slicers hangs while a job is scheduling
Hitting /txt/slicers while a job is in the process of scheduling causing the endpoint to hang and never return.

### PR DESCRIPTION
**Pull request by jsnoble**
…#589

### CODE REVIEW

File: lib/cluster/services/api.js [hunk 1]
```diff
@@ -465,8 +488,15 @@ module.exports = function module(context, app) {
...
-                sendError(res, err.code, err.message);
+            .catch((errObj) => {
+                if (errObj.code) {
+                    logger.error(errObj.message);
+                    sendError(res, errObj.code, errObj.message);
+                    return;
+                }
+                const errMsg = `could not get slicer statistics, error: ${parseError(errObj)}`;
+                logger.error(errMsg);
+                sendError(res, 500, errMsg);
```
Comment (kstaken) [inline lib/cluster/services/api.js]: Come on ... this code sequence is repeated 4 times... I can't explain how frustrating that is after all the time we spent trying to clean this module up. 

Does this really require special casing like this? Where does that `.code` attribute come from? Why can't it simply be handled in the general error handler?

File: lib/cluster/services/cluster/backends/native.js [hunk 2]
```diff
@@ -601,22 +601,20 @@ module.exports = function module(context, messaging, executionService) {
         }));
     }
 
-    function getSlicerStats(exId) {
+    function getSlicerStats(exIds, specificId) {
```
Comment (kstaken) [inline lib/cluster/services/cluster/backends/native.js]: This looks questionable to me. What required this API to change in this way?

**Comment by godber** [comment]
I am not really seeing #659 as being resolved.

When I run the integration tests, I have typically been able to run `tstop -p 45678` and watch the jobs progress ... now it appears to hang.  Checking with curl it appears that `/txt/slicers` blocks while a job is running and `/txt/workers` and `/txt/nodes`.

Pretty easy to reproduce ... just start up the integration tests and walk through the endpoints ... 

```
curl -Ss localhost:45678/txt/workers
curl -Ss localhost:45678/txt/jobs
curl -Ss localhost:45678/txt/ex
curl -Ss localhost:45678/txt/slicers
```

They will all return except slicers.  You might catch the tests between jobs and `/txt/slicers` returns then ... but its not hard to get it to fail.

**Comment by kstaken** [comment]
I talked to @jsnoble about this and he thinks the issue with hitting it while the integration tests are running is down to the rapid fire nature of how jobs are scheduled in the tests and tstop catching them after a job has ended but before the slicer has fully shutdown. This is a different issue and he will open a separate ticket and put a fix in separate PR. 

As for #659 this PR does appear to resolve that issue. The integration tests pass otherwise so going ahead with a merge.

**Pull request merged**